allennlp/models/coreference_resolution/coref.py

```diff
@@ -111,8 +111,8 @@ def __init__(self,
     def forward(self,  # type: ignore
                 text: Dict[str, torch.LongTensor],
                 spans: torch.IntTensor,
-                span_labels: torch.IntTensor = None,
-                metadata: List[Dict[str, Any]] = None) -> Dict[str, torch.Tensor]:
+                metadata: List[Dict[str, Any]],
```

Comment:
My first thought was that changing the order of this was a problem for backwards compatibility, but because it's a `Model.forward` method, we'll pass a dictionary of parameters, anyway, so this should probably be ok.

Reply:
Actually, looking at this again, `metadata` is in fact optional (e.g., it's not supplied in our demo), so there is no need to change this.

Reply:
Yeah, I thought about the same. Maybe warn in the next release, cause someone may unlikely but possibly pass params to a model like this one in order.

Reply:
So you're saying that I should change `metadata` back to be optional? I see that's needed when `span_labels` is supplied (see the code at the end of this method). What should I do then? Just keep it optional, as it was?

Reply:
The intent of most of these models is that you have a "training" setting where you have labels provided, and a "testing" setting where there are no labels.  `metadata` is only needed in the "training" setting, where you have `span_labels` given.  So, yes, it is optional.